detd/manager.py

```diff
@@ -96,12 +96,17 @@ def add_talker(self, config):
 
 class InterfaceManager():
 
-    def __init__(self, interface):
+    def __init__(self, interface, options):
 
         logger.info(f"Initializing {__class__.__name__}")
 
         self.interface = interface
-        self.mapping = Mapping(self.interface)
+        self.options = options
+        
+        if self.options.qdiscmap == "nomap":
```

Comment:
This will be replaced by a selection of different mapping classes based on the value of the options.

For software based taprio, most likely we can use the MappingNaive (https://github.com/Avnu/detd/blob/1a327a11d8a8c6ec9a57940063421e03d641db43/detd/mapping.py#L33). But we will need to check.

The idea is that we do not need to pass the options further down in the framework. The initialization will pick up the specific classes to implement the backend specification.